app/controllers/DataFrameUtils.scala

```diff
@@ -0,0 +1,74 @@
+package controllers
+
+import controllers.EnforcementEngineResponseProcessor.debugMode
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.slf4j.LoggerFactory
+import play.api.Configuration
+
+object DataFrameUtils {
+  private val LOGGER = LoggerFactory.getLogger("DataFrameUtils")
+
+  def anyNotNull(row: Row, columnName: String = Constants.SUBJECT_ID_COL_NAME): Boolean = {
+    val len = row.length
+
+    var i = 0
+    var fieldNames = row.schema.fieldNames
+    //print patientId if its the only col
+    if (len == 1 && fieldNames(0).equals(columnName))
+      return true
+    //skip patientId
+    while (i < len) {
```

Comment:
Better use for loop, instead of distributing this i initialization and checks and increment